addons/account/models/account_payment.py

```diff
@@ -116,7 +116,7 @@ class AccountPayment(models.Model):
         comodel_name='res.partner',
         string="Customer/Vendor",
         ondelete='restrict',
-        domain="['|', ('parent_id','=', False), ('is_company','=', True)]",
+        domain="['|', ('parent_id','=', False), ('vat','!=', False)]",
```

Comment:
The domain here and the one in account_bank_statement_line.py have to be *exactly* the same like all the domain you are changing